pandas/_libs/tslib.pyx

```diff
@@ -561,6 +600,24 @@ cpdef array_to_datetime(
         else:
             tz_offset = out_tzoffset_vals.pop()
             tz_out = timezone(timedelta(seconds=tz_offset))
+
+    if infer_reso:
+        if state.creso_ever_changed:
+            # We encountered mismatched resolutions, need to re-parse with
```

Comment:
Probably misreading but in the case of having to do a second pass how do we establish what the `correct` creso is? Just whatever came last?

Reply:
we use the highest-detected resolution